### PR TITLE
Separate lab config from server launch config

### LINKED ISSUE #1552: ContentsManager.untitled_notebook sometime ignored.
Lab does not seem to respect `FileContentsManager.untitled_notebook`. Let say, – For example:

```
$ cat ~/.jupyter/jupyter_notebook_config.py
c.FileContentsManager.untitled_notebook='A Notebook has No Name - '
```

Classic notebook will create (`A Notebook has No Name - 1.ipynb` ...). Lab (still) create `Untilted1.ipynb`.

Arguably not such a used feature. Astonishingly if you start with `jupyter lab` and manually access `/tree` notebook also does not respect it...

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1552.

The lab config is now added as a separate entry in the `jupyter_notebook_config.json` file.  We use the notebook app to fetch the appropriate config for our server extension.

cc @Carreau @jasongrout @SylvainCorlay

### CODE REVIEW

**Comment by blink1073** [comment]
Heads up @gnestor, this will require labextensions to be re-enabled after the change.

**Comment by minrk** [comment]
Relevant to https://github.com/jupyter/notebook/issues/2063#issuecomment-276476964, I think lab extensions may want to not use traitlets or traitlets.config at all, as they don't allow the appropriate merge semantics you want for enabling extensions at different levels. Following the behavior of `nbconfig` (frontend extension config in the notebook app), which does a deep, recursive merge (as opposed to only at the class level).

Basically, follow what `jupyter nbextension` does, not what `jupyter serverextension` does.
